src/main/java/app/model/agents/AgentImp.java

```diff
@@ -33,6 +33,9 @@ public class AgentImp implements Agent
     @Getter protected VectorSet seen;
     @Getter protected AgentView agentViewWindow;
     protected final boolean DRAW_HEARD = false;
+    @Getter protected boolean solid;
+    @Getter protected boolean visible;
+    @Getter protected boolean transparent;
```

Comment:
Rather than initialising these three variables three times over in three different constructors, why not just initialise them at  creation time?
```java
 @Getter protected boolean solid = true;
 ```

Reply:
I'll set some default values yeah, since mose are solid, visible and not transparent.